### PR TITLE
[Selenium] Create test for check git with self sign certs

### PR DESCRIPTION
**Pull request by SkorikSergey**
### What does this PR do?
We had customer issue where used self sign certs. On this moment we are checking git features using third part service (github). We can create deployment with git service. For this good fit gogs git (https://gogs.io/) with add self-sign cert.

This test was used for checking CRW 2.10.0 from `7.32.x` branch. Test result - https://main-jenkins-csb-crwqe.apps.ocp4.prod.psi.redhat.com/job/CodereadyWorkspaces/job/e2e/job/basic/job/typescript-tests/8936/

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/17288

### CODE REVIEW

File: tests/e2e/tests/e2e/GitSelfSignCert.spec.ts [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+ });
+
+ suite('Stopping and deleting the workspace', async () => {
+    test(`Stop and remove workspace`, async () => {
+        await workspaceHandlingTests.stopAndRemoveWorkspace(workspaceName);
+    });
+});
+
+ async function cloneTestRepo(linkToRepo: string) {
```
Comment (Katka92) [inline tests/e2e/tests/e2e/GitSelfSignCert.spec.ts:96]: Would it be useful also for some other test or is it specific for just this one test case? If another test could benefit from this method, I would suggest moving it to some library. If it's specific to just this test case, I'm OK leaving it here.

File: tests/e2e/TestConstants.ts [hunk 2]
```diff
@@ -254,6 +254,10 @@ export const TestConstants = {
     /**
      * This variable determines whether to delete the workspace after the test or leave the workspace running.
      */
-    TS_DELETE_PLUGINS_TEST_WORKSPACE: process.env.TS_DELETE_PLUGINS_TEST_WORKSPACE || 'true'
+    TS_DELETE_PLUGINS_TEST_WORKSPACE: process.env.TS_DELETE_PLUGINS_TEST_WORKSPACE || 'true',
 
+    /**
+     * URL of the workspace created by devworkspace-controller.
```
[the remark this answers is not in the document]
Comment (SkorikSergey) [reply tests/e2e/TestConstants.ts]: You right. Thank you. It will fix it.

[the remark this answers is not in the document]
Comment (Katka92) [reply tests/e2e/TestConstants.ts]: Thx

File: tests/e2e/tests/e2e/GitSelfSignCert.spec.ts (hunk 1 above)
Comment (SkorikSergey) [reply tests/e2e/tests/e2e/GitSelfSignCert.spec.ts:96]: Very similar function exists in GitSsh and GitPublishBranch tests. But for GitSelfSignCert cloneTestRepo() is specific in getting repo url step.

**Review by Katka92 — APPROVED** [review]

**Review by Ohrimenko1988 — APPROVED** [review]

**Pull request merged**